edb/pgsql/compiler/relctx.py

```diff
@@ -69,12 +74,18 @@ def pull_path_namespace(
 
     for source_q in source_qs:
         s_paths: Set[Tuple[irast.PathId, str]] = set()
-        if hasattr(source_q, 'path_outputs'):
-            s_paths.update(source_q.path_outputs)
-        if hasattr(source_q, 'path_namespace'):
-            s_paths.update(source_q.path_namespace)
-        if isinstance(source_q, pgast.Query):
-            s_paths.update(source_q.path_rvar_map)
+        if flavor == 'normal':
```

Comment:
Please use an explicit list of arms for `flavor` here with an `AssertionError` in else in case we add more of them in the future.